### ISSUE TITLE
Chip Classification: Support Stochastic Gradient Descent

### ISSUE THREAD

**Issue by jamesmcclain** [azavea/raster-vision#790]
Currently, only the ADAM optimizer is offered.  Recent thought seems to point to it having a tendency to become stuck in local minima.  It converges more quickly than SGD, leading to faster training, but may produce inferior results.
